.github/pull_request_template.md

```diff
@@ -1,18 +1,21 @@
 <!--
-Thank you for your pull request.
+Thank you for your pull request!
+
 Please see http://site.icu-project.org/processes/contribute for general
 information on contributing to ICU.
 
-You will be automatically asked to sign the contributors license before the PR is accepted.
+You will be automatically asked to sign the contributors license (CLA) before the PR is accepted.
 - sign: https://cla-assistant.io/unicode-org/icu
 - license: http://www.unicode.org/copyright.html
+
+Please note that all commit messages must be prefixed with a JIRA Issue number.
+For example: "ICU-1234 Fix xyz"
 -->
 
 ##### Checklist
 
 - [ ] Issue filed: https://unicode-org.atlassian.net/browse/ICU-_____
-- [ ] Updated PR title and link in previous line to include Issue number
+- [ ] Commit message(s) prefixed with Issue number, updated PR title and link above to include Issue number
 - [ ] Issue accepted
 - [ ] Tests included
 - [ ] Documentation is changed or added
```

Comment:
How about
```
- [ ] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-_____
- [ ] Required: The PR title must be prefixed with a JIRA Issue number. For example: "ICU-1234 Fix xyz"
- [ ] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [ ] Required: Each commit message must be prefixed with a JIRA Issue number. For example: "ICU-1234 Fix xyz"
- [ ] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable
```

Reply:
Oh, this is much better. :)

I wasn't sure about using stronger language in the checklist, but on reflection I think that it's actually better to be explicit for these since they are indeed required.

I'll force-push in a moment, so as to avoid the round-trip for squashing...